docs/ocean/design_docs/min_level_cell.rst

```diff
@@ -0,0 +1,769 @@
+   remains exactly zero.
+
+2. ``z_level_full_cells`` - S is horizontally uniform (T remains uniform in
+   3D), stably stratified in vertical. Disable vertical tracer mixing. Full top
+   and bottom cells. Verify T and S  remain unchanged, velocity remains exactly
+   zero.
+
+3. ``z_level`` - Same as 2. but partial top and bottom cells. Maximum velocity
+   is a measure of the error and should grow slowly at top and bottom(e.g.
+   1e-6m/s after a day).
+
+To begin with, these tests also have melt fluxes disabled along with all
+tendency terms.  As part of debugging, we will gradually turn on tendencies
+and check the behavior.
+
+We will run on multiple machines (Ubuntu laptop, Anvil, Grizzly, Cori, Compy)
+with a mix of Gnu and Intel compilers.  We will plot the resulting T, S and KE
+fields to make sure they look comparable to the results with the current
+terrain-following coordinate.
+
+We will test with JM and Wright equations of state.
+
+Testing: Surface fields and fluxes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/02/11
+
+Contributors: Xylar Asay-Davis
+
+The testing in ``sub_ice_shelf_2D`` will account for the surface fluxes and
+fields that we most anticipate being affected by ``minLevel*``.  Testing of
+other surface fields will likely require running tests that include GM, KPP
+and other parameterizations that are not part of this configuration by default.
+
+Testing: Correct boundary conditions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/02/25
+
+Contributors: Mark Petersen and Xylar Asay-Davis
+
+The 3 z-level versions of ``sub_ice_shelf_2D`` described above should provide
+sufficient testing of the boundary conditions related to tendency terms (e.g.
+higher-order tracer reconstruction).  More sophisticated parameterizations
+(KPP and GM) will be addressed in the second phase of this work.
+
+Testing: Inactivate cells during simulation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/02/11
+
+Contributors: Xylar Asay-Davis
+
+Since this is just a consideration, no testing is necessary.  We will add
+timers around ``ocn_init_routines_compute_min_max_level()`` (if they are not
+already present) to see if the timing is significant, though a global test
+may be necessary to get a realistic feel.
+
+Testing: Bit-for-bit when no inactive top cells
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/02/11
+
+Contributors: Xylar Asay-Davis
+
+We will run the legacy COMPASS nightly and land-ice-fluxes regression suites
+on multiple machines (Ubuntu laptop, Anvil, Grizzly, Compy) with a mix of Gnu
+and Intel compilers to make sure nothing has changed when we run with
+``minLevelCell = 1`` everywhere.
+
+We will also run the following E3SM tests before and after the changes:
+
+* ``SMS.T62_oQU120_ais20.MPAS_LISIO_TEST.<<machine>>_intel``
+* ``SMS.T62_oQU120_ais20.MPAS_LISIO_TEST.<<machine>>_gnu``
```

Comment:
These tests didn't catch https://github.com/E3SM-Project/E3SM/issues/4152 so we probably want at least one additional (`oECv3`?) test against a baseline.